src/server/engine_shard_set.cc

```diff
@@ -70,6 +79,66 @@ ShardMemUsage ReadShardMemUsage(float wasted_ratio) {
   return usage;
 }
 
+// TODO: document this class.
+// Thread safe.
+class RoundRobinSharder {
+ public:
+  void InitThreadLocal() {
+    round_robin_prefix_ = absl::GetFlag(FLAGS_round_robin_prefix);
+
+    if (IsEnabled()) {
+      // 500k entries will consume 1mb per thread, and will allow 100 keys with < 1% collision
+      // probability. Since this has a considerable footprint, we only allocate when enabled.
+      constexpr size_t kRoundRobinSize = 500'000;
+      round_robin_shards_.resize(kRoundRobinSize);
+      std::fill(round_robin_shards_.begin(), round_robin_shards_.end(), kInvalidSid);
+    }
+  }
```

Comment:
So this is by the birthday paradox? Maybe we actually don't need such strong guarantees and should look when does the distribution as a whole become out of balance because of it. Because 500k for 100 keys is _a lot_

Reply:
500k is indeed much higher than I'd have wished.
What would you define to be "out of balance"?
Anyway, given this feature will be mostly disabled I'm not too worried about this.
(btw while it's 500k _entries_, each entry is `ShardId` which is 2 bytes, so really this is 1mb per thread)

Reply:
Also true, lets think about optimizations later. We can then possibly reduce the size and use 1 byte per shard for example

Reply:
I thought about using 1 byte and `CHECK()`ing that shards are indeed small enough, but that seemed odd to me given we define `ShardId` to be 2 bytes.
Anyway, with 100k entries (200kb per-thread overhead) we have 2.5% chance of collisions with 100 keys, I think that's pretty ok still.

Reply:
It's a little odd but I really doubt someone uses over 255 shards 🙂

Reply:
Agreed, but my thinking was that this should be a restriction (or decision) for all of Dragonfly, not just this feature..